### PR TITLE
Simplify Dashboard UX via `zenml.show()`

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes

I added a `zenml.show()` Python function and corresponding `zenml show` CLI command that can be used to open the ZenML dashboard in an environment-specific way:
- In the native environment, the dashboard is opened in a new browser tab (similar as `zenml up` does right now)
- In Colab and other notebooks, the dashboard is embedded as an IFrame
- To work with Colab, one can also provide a `zenml.show(NGROK_TOKEN)` arg to automatically tunnel the dashboard via ngrok.

### New UX in notebooks:
Screenshot 2023-04-28 at 14 32 01

### Old UX in notebooks:
Screenshot 2023-04-28 at 14 14 29

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by fa9r** [comment]
Some inspiration from [gradio docs](https://gradio.app/docs/#interface-launch):

Screenshot 2023-05-02 at 12 04 58

**Review by htahir1 — COMMENTED** [review]
In general, I like this UX a lot! Now sure what the value of having zenml.up() zenml.down() would be though? Do you mean they wouldnt have to do `!zenml up` in the notebook?

File: src/zenml/cli/server.py [hunk 1]
```diff
@@ -260,26 +226,29 @@ def up(
                 )
 
             if server.status and server.status.url:
-                cli_utils.declare(
-                    f"The local ZenML dashboard is available at "
+                logger.info(
+                    f"The ZenML dashboard is available at "
```
Comment (fa9r) [inline src/zenml/cli/server.py:230]:
```suggestion
                cli_utils.declare(
                    f"The local ZenML dashboard is available at "
```

File: src/zenml/cli/server.py [hunk 2]
```diff
@@ -260,26 +226,29 @@ def up(
...
                     f"'{server.status.url}'. You can connect to it using the "
                     f"'{DEFAULT_USERNAME}' username and an empty password. "
-                    f"To open the dashboard in a browser automatically, "
-                    f"set the env variable AUTO_OPEN_DASHBOARD=true."
                 )
+                zenml.show()
```
Comment (fa9r) [inline src/zenml/cli/server.py:234]:
```suggestion
                zenml.show(ngrok=ngrok)
```

File: src/zenml/utils/dashboard_utils.py [hunk 3]
```diff
@@ -81,3 +83,42 @@ def print_run_url(run_name: str, pipeline_id: Optional[UUID] = None) -> None:
...
+        from IPython.display import IFrame
+
+        display(IFrame(src=url, width="100%", height=720))
+
+    elif environment == EnvironmentType.NATIVE:
+        if handle_bool_env_var(ENV_AUTO_OPEN_DASHBOARD, default=True):
+            try:
+                import webbrowser
+
+                webbrowser.open(url)
```
Comment (htahir1) [inline src/zenml/utils/dashboard_utils.py:109]: This webbrowser.open doesnt work on WSL. I wanted to fix that for a while, but there  is no real solution except here: https://github.com/python/cpython/issues/89752#issuecomment-1254943311 . Maybe we can try this suggestion in a try block?

```python
import os
import shutil


## maybe add these where we detect the OS to add another WSL type
from platform import uname

# source: https://www.scivision.dev/python-detect-wsl/
def in_wsl() -> bool:
    return 'microsoft-standard' in uname().release
    
# probably wrap in a try catch
if in_wsl():
    webbrowser.get("wslview %s").open(url)
```

**Review by htahir1 — CHANGES_REQUESTED** [review]
Just left one comment, for the other part I am not the best person to review

File: src/zenml/cli/server.py [hunk 4]
```diff
@@ -128,6 +85,12 @@ def get_active_deployment(local: bool = False) -> Optional["ServerDeployment"]:
     help="Use a custom Docker image for the ZenML server. Only used when "
     "`--docker` is set.",
 )
+@click.option(
+    "--ngrok",
+    type=str,
+    default=None,
+    help="Specify an ngrok auth token to use for exposing the ZenML server.",
+)
```
Comment (AlexejPenner) [inline src/zenml/cli/server.py:93]: How about we call this auth-token or something more obvious to the user?

**Review by AlexejPenner — APPROVED** [review]
Generally looks good, lets get this in as well to the next release. I guess the cahnges on the quickstart notebook are probably deprecated though

File: src/zenml/cli/server.py (hunk 4 above)
Comment (fa9r) [reply src/zenml/cli/server.py:93]: Makes sense I renamed this to `ngrok-token` and also renamed `ngrok` args to `ngrok_token` everywhere

File: src/zenml/utils/dashboard_utils.py (hunk 3 above)
Comment (fa9r) [reply src/zenml/utils/dashboard_utils.py]: Ok dashboard opening should work under WSL, could you give it a try and see if it works?

**Review by htahir1 — APPROVED** [review]
Ok makes sense!

File: src/zenml/utils/dashboard_utils.py (hunk 3 above)
Comment (fa9r) [reply src/zenml/utils/dashboard_utils.py]: Update: worked! 🎉

**Pull request merged**